Enable multibuilld and create test flavor

ycedres opened the pull request:
With this PR we enable `multibuild` in order to create a repository that includes the `python3-salt-testsuite` package and another repository without this `python3-salt-testsuite` package  so we can submit it to `openSUSE:Factory` without having to include more dependencies to Ring 1.


Also, in the context of pushing `python3-salt-testsuite` to Factory and since the submission of  `python-mock` was rejected, we need to disable this dependency. An alternative strategy was followed, favouring the use of `unittest.mock` for newer Python versions. See this [PR](https://github.com/openSUSE/salt/pull/612).

meaksh reviewed (changes requested):
Hmm, I don't think this is going to work. Have you tested this in a an OBS branch?

IIUC you cannot use `<` to do string comparison: https://rpm-software-management.github.io/rpm/manual/spec.html

Maybe it is better to use `%{suse_version}` instead of `%{python_flavor}`?: https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto

BTW, you have a typo in the commit message: "Re**r**quire"

agraul commented:
If we were to use the proper macros, we could do a version comparison inside `%{python_module}` (see https://github.com/openSUSE/python-rpm-macros#macros for reference)

ycedres committed:
Require python-mock for older Python versions

ycedres committed:
Enable multibuild to create test flavor

The goal is to build a repository that includes the python3-salt-testsuite package and another repository without it so we can submit to openSUSE:Factory without having to include more dependencies to Ring 1.

ycedres commented:
At the moment of writing this comment, you can see that this is building the two repositories here:

https://build.opensuse.org/package/show/home:ygutierrez:branches:systemsmanagement:saltstack:products:testing-mock-tw/salt

m-czernek commented on salt/_multibuild:
```diff
@@ -0,0 +1,3 @@
+<multibuild>
+  <flavor>test</flavor>
```
Could we make the flavor name more descriptive? E.g.:

```suggestion
  <flavor>testsuite_included</flavor>
```

(Not sure if there's an RPM convention though, to include flavor names such as `test`/`debug`/..., in which case feel free to disregard)

agraul replied:
`test` is often used to enable running tests in `%check` , e.g. in https://build.opensuse.org/package/show/openSUSE:Factory/rust1.74.

m-czernek reviewed (approved):
A small comment, otherwise LGTM

ycedres committed:
Rename test flavor denomination

agraul reviewed (approved):
LGTM, let's give it a try!

meaksh reviewed (approved):
Thanks for the adjustments, let's give it a try!
